### ISSUE TITLE
RPi 2B Rev 1.1 no longer boots when no monitor is attached via HDMI

### ISSUE THREAD

**Issue by Hiradur** [raspberrypi/linux#4506]
**Describe the bug**
After a Kernel update via the `raspberrypi-kernel` package to version `5.10.52-v7+ #1440 SMP Tue Jul 27 09:54:13 BST 2021 armv7l GNU/Linux` package, my headless RPi 2B server no longer booted when no monitor was attached via HDMI. Initially, the activity LED flickers showing that the Pi is doing something but it does not boot far enough to be ping-able or accessible via SSH. I commented out `gpu_mem` in `/boot/config.txt` as suggested elsewhere but that didn't help.

Either one of the following makes it boot again:
- Attach a monitor via HDMI and reboot
- Downgrade firmware to `e7785df5` via `rpi-update`

**To reproduce**
- Use RPi2 2B Rev 1.1
- Upgrade Kernel to version `5.10.52-v7+ #1440 SMP Tue Jul 27 09:54:13 BST 2021 armv7l GNU/Linux` via `raspberrypi-kernel` package. Alternatively, the issue is also present with firmware version `d4c184e19b8b48ac358ae595d09db1d77bf0f150`
- Disconnect any monitors
- Power on the Pi
- Try to `ping` it or `ssh` into it

**Expected behaviour**
RPi 2B boots up and is accessible via SSH without a monitor connected via HDMI.

**Actual behaviour**
RPi 2B does not boot successfully and is neither ping-able nor accessible via SSH without a monitor connected via HDMI.

**System**
This is the output of `raspinfo` with a monitor connected via HDMI and a firmware version which causes the boot issue when no monitor is connected since I cannot access this information otherwise:
```
tee: raspinfo.txt: Permission denied
System Information
------------------

Raspberry Pi 2 Model B Rev 1.1
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"

Raspberry Pi reference 2018-11-13
Generated using pi-gen, https://github.com/RPi-Distro/pi-gen, 7e0c786c641ba15990b5662f092c106beed40c9f, stage2

Linux server 5.10.52-v7+ #1441 SMP Tue Aug 3 18:10:09 BST 2021 armv7l GNU/Linux
Revision	: a01041
Serial		: redacted
Model		: Raspberry Pi 2 Model B Rev 1.1
VCHI initialization failed
Throttled flag  : 
VCHI initialization failed
Camera          : 

Videocore information
---------------------

VCHI initialization failed

VCHI initialization failed

Filesystem information
----------------------
Filesystem     1K-blocks    Used Available Use% Mounted on
/dev/root       15026972 3003220  11380628  21% /
devtmpfs          340908       0    340908   0% /dev
tmpfs             472492       0    472492   0% /dev/shm
tmpfs             472492    6312    466180   2% /run
tmpfs               5120       4      5116   1% /run/lock
tmpfs             472492       0    472492   0% /sys/fs/cgroup
/dev/mmcblk0p1    257087   49192    207896  20% /boot
tmpfs              94496       0     94496   0% /run/user/1002

Filename				Type		Size		Used		Priority
/var/swap                               file		102396		0-2

Package version information
---------------------------
raspberrypi-ui-mods:
  Installed: (none)
raspberrypi-sys-mods:
  Installed: 20210706
openbox:
  Installed: (none)
lxpanel:
  Installed: (none)
pcmanfm:
  Installed: (none)
rpd-plym-splash:
  Installed: (none)

Networking Information
----------------------

eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet x.x.x.x  netmask x.x.x.x  broadcast x.x.x.x
        inet6 y::y.y.y.y  prefixlen 64  scopeid 0x20<link>
        inet6 y.y.y.y.y.y.y.y  prefixlen 64  scopeid 0x0<global>
        ether m.m.m.m  txqueuelen 1000  (Ethernet)
        RX packets 435  bytes 34861 (34.0 KiB)
        RX errors 0  dropped 2  overruns 0  frame 0
        TX packets 231  bytes 63018 (61.5 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet x.x.x.x  netmask x.x.x.x
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0


USB Information
---------------

/:  Bus 01.Port 1: Dev 1, Class=root_hub, Driver=dwc_otg/1p, 480M
    |__ Port 1: Dev 2, If 0, Class=Hub, Driver=hub/5p, 480M
        |__ Port 1: Dev 3, If 0, Class=Vendor Specific Class, Driver=smsc95xx, 480M

config.txt
----------

VCHI initialization failed
VCHI initialization failed

cmdline.txt
-----------
coherent_pool=1M snd_bcm2835.enable_compat_alsa=0 snd_bcm2835.enable_hdmi=1 video=HDMI-A-1:1920x1080M@60,margin_left=48,margin_right=48,margin_top=48,margin_bottom=48 vc_mem.mem_base=0x3ec00000 vc_mem.mem_size=0x40000000  dwc_otg.lpm_enable=0 console=tty1 root=PARTUUID=6e8533f3-02 rootfstype=ext4 elevator=deadline fsck.repair=yes rootwait

raspi-gpio settings
-------------------

Must be root

vcdbg log messages
------------------

Failed to allocate -385826387 bytes for message buffer

dmesg log
---------

[    0.000000] Booting Linux on physical CPU 0xf00
[    0.000000] Linux version 5.10.52-v7+ (dom@buildbot) (arm-linux-gnueabihf-gcc-8 (Ubuntu/Linaro 8.4.0-3ubuntu1) 8.4.0, GNU ld (GNU Binutils for Ubuntu) 2.34) #1441 SMP Tue Aug 3 18:10:09 BST 2021
[    0.000000] CPU: ARMv7 Processor [410fc075] revision 5 (ARMv7), cr=10c5387d
[    0.000000] CPU: div instructions available: patching division code
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT aliasing instruction cache
[    0.000000] OF: fdt: Machine model: Raspberry Pi 2 Model B Rev 1.1
[    0.000000] random: fast init done
[    0.000000] Memory policy: Data cache writealloc
[    0.000000] Reserved memory: created CMA memory pool at 0x1ec00000, size 256 MiB
[    0.000000] OF: reserved mem: initialized node linux,cma, compatible id shared-dma-pool
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000000000000-0x000000003b3fffff]
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x000000003b3fffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x000000003b3fffff]
[    0.000000] On node 0 totalpages: 242688
[    0.000000]   DMA zone: 2133 pages used for memmap
[    0.000000]   DMA zone: 0 pages reserved
[    0.000000]   DMA zone: 242688 pages, LIFO batch:63
[    0.000000] percpu: Embedded 20 pages/cpu s50700 r8192 d23028 u81920
[    0.000000] pcpu-alloc: s50700 r8192 d23028 u81920 alloc=20*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1 [0] 2 [0] 3 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 240555
[    0.000000] Kernel command line: coherent_pool=1M snd_bcm2835.enable_compat_alsa=0 snd_bcm2835.enable_hdmi=1 video=HDMI-A-1:1920x1080M@60,margin_left=48,margin_right=48,margin_top=48,margin_bottom=48 vc_mem.mem_base=0x3ec00000 vc_mem.mem_size=0x40000000  dwc_otg.lpm_enable=0 console=tty1 root=PARTUUID=6e8533f3-02 rootfstype=ext4 elevator=deadline fsck.repair=yes rootwait
[    0.000000] Kernel parameter elevator= does not have any effect anymore.
               Please use sysfs to set IO scheduler for individual devices.
[    0.000000] Dentry cache hash table entries: 131072 (order: 7, 524288 bytes, linear)
[    0.000000] Inode-cache hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 681816K/970752K available (10240K kernel code, 1312K rwdata, 2948K rodata, 1024K init, 863K bss, 26792K reserved, 262144K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] ftrace: allocating 32043 entries in 63 pages
[    0.000000] ftrace: allocated 63 pages with 6 groups
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] 	Rude variant of Tasks RCU enabled.
[    0.000000] 	Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
[    0.000000] NR_IRQS: 16, nr_irqs: 16, preallocated irqs: 16
[    0.000000] random: get_random_bytes called from start_kernel+0x3ac/0x580 with crng_init=1
[    0.000000] arch_timer: cp15 timer(s) running at 19.20MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x46d987e47, max_idle_ns: 440795202767 ns
[    0.000011] sched_clock: 56 bits at 19MHz, resolution 52ns, wraps every 4398046511078ns
[    0.000035] Switching to timer-based delay loop, resolution 52ns
[    0.000419] Console: colour dummy device 80x30
[    0.001421] printk: console [tty1] enabled
[    0.001517] Calibrating delay loop (skipped), value calculated using timer frequency.. 38.40 BogoMIPS (lpj=192000)
[    0.001591] pid_max: default: 32768 minimum: 301
[    0.001888] LSM: Security Framework initializing
[    0.002246] Mount-cache hash table entries: 2048 (order: 1, 8192 bytes, linear)
[    0.002328] Mountpoint-cache hash table entries: 2048 (order: 1, 8192 bytes, linear)
[    0.004375] cgroup: Disabling memory control group subsystem
[    0.004837] CPU: Testing write buffer coherency: ok
[    0.005669] CPU0: update cpu_capacity 1024
[    0.005724] CPU0: thread -1, cpu 0, socket 15, mpidr 80000f00
[    0.007632] Setting up static identity map for 0x100000 - 0x10003c
[    0.007952] rcu: Hierarchical SRCU implementation.
[    0.009315] smp: Bringing up secondary CPUs ...
[    0.011356] CPU1: update cpu_capacity 1024
[    0.011370] CPU1: thread -1, cpu 1, socket 15, mpidr 80000f01
[    0.013357] CPU2: update cpu_capacity 1024
[    0.013372] CPU2: thread -1, cpu 2, socket 15, mpidr 80000f02
[    0.015255] CPU3: update cpu_capacity 1024
[    0.015267] CPU3: thread -1, cpu 3, socket 15, mpidr 80000f03
[    0.015505] smp: Brought up 1 node, 4 CPUs
[    0.015692] SMP: Total of 4 processors activated (153.60 BogoMIPS).
[    0.015731] CPU: All CPU(s) started in HYP mode.
[    0.015763] CPU: Virtualization extensions available.
[    0.017284] devtmpfs: initialized
[    0.039057] VFP support v0.3: implementor 41 architecture 2 part 30 variant 7 rev 5
[    0.039533] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.039618] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.055463] pinctrl core: initialized pinctrl subsystem
[    0.057329] NET: Registered protocol family 16
[    0.062701] DMA: preallocated 1024 KiB pool for atomic coherent allocations
[    0.070343] audit: initializing netlink subsys (disabled)
[    0.071536] audit: type=2000 audit(0.070:1): state=initialized audit_enabled=0 res=1
[    0.071899] thermal_sys: Registered thermal governor 'step_wise'
[    0.073083] hw-breakpoint: found 5 (+1 reserved) breakpoint and 4 watchpoint registers.
[    0.073168] hw-breakpoint: maximum watchpoint size is 8 bytes.
[    0.073621] Serial: AMBA PL011 UART driver
[    0.100586] bcm2835-mbox 3f00b880.mailbox: mailbox enabled
[    0.120519] raspberrypi-firmware soc:firmware: Attached to firmware from 2021-08-03T18:17:07, variant start
[    0.130531] raspberrypi-firmware soc:firmware: Firmware hash is 40787ee5905644f639a2a0f6e00ae12e517a2211
[    0.185651] Kprobes globally optimized
[    0.193380] bcm2835-dma 3f007000.dma: DMA legacy API manager, dmachans=0x1
[    0.196594] SCSI subsystem initialized
[    0.197018] usbcore: registered new interface driver usbfs
[    0.197141] usbcore: registered new interface driver hub
[    0.197271] usbcore: registered new device driver usb
[    0.200010] clocksource: Switched to clocksource arch_sys_counter
[    1.947152] VFS: Disk quotas dquot_6.6.0
[    1.947377] VFS: Dquot-cache hash table entries: 1024 (order 0, 4096 bytes)
[    1.947698] FS-Cache: Loaded
[    1.948157] CacheFiles: Loaded
[    1.949516] simple-framebuffer 3e513000.framebuffer: framebuffer at 0x3e513000, 0x6e7000 bytes, mapped to 0x(ptrval)
[    1.949600] simple-framebuffer 3e513000.framebuffer: format=a8r8g8b8, mode=1824x984x32, linelength=7296
[    1.984339] Console: switching to colour frame buffer device 228x61
[    2.016214] simple-framebuffer 3e513000.framebuffer: fb0: simplefb registered!
[    2.031069] NET: Registered protocol family 2
[    2.031661] IP idents hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    2.034324] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    2.034868] TCP established hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    2.035255] TCP bind hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    2.035685] TCP: Hash tables configured (established 8192 bind 8192)
[    2.036185] UDP hash table entries: 512 (order: 2, 16384 bytes, linear)
[    2.036496] UDP-Lite hash table entries: 512 (order: 2, 16384 bytes, linear)
[    2.037756] NET: Registered protocol family 1
[    2.039220] RPC: Registered named UNIX socket transport module.
[    2.039453] RPC: Registered udp transport module.
[    2.039624] RPC: Registered tcp transport module.
[    2.039794] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    2.042144] hw perfevents: enabled with armv7_cortex_a7 PMU driver, 5 counters available
[    2.047805] Initialise system trusted keyrings
[    2.048529] workingset: timestamp_bits=14 max_order=18 bucket_order=4
[    2.061565] zbud: loaded
[    2.064617] FS-Cache: Netfs 'nfs' registered for caching
[    2.066161] NFS: Registering the id_resolver key type
[    2.066429] Key type id_resolver registered
[    2.066589] Key type id_legacy registered
[    2.066975] nfs4filelayout_init: NFSv4 File Layout Driver Registering...
[    2.067216] nfs4flexfilelayout_init: NFSv4 Flexfile Layout Driver Registering...
[    2.069196] Key type asymmetric registered
[    2.069378] Asymmetric key parser 'x509' registered
[    2.078904] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 249)
[    2.088129] io scheduler mq-deadline registered
[    2.097368] io scheduler kyber registered
[    2.117822] bcm2835-rng 3f104000.rng: hwrng registered
[    2.127715] vc-mem: phys_addr:0x00000000 mem_base=0x3ec00000 mem_size:0x40000000(1024 MiB)
[    2.138226] gpiomem-bcm2835 3f200000.gpiomem: Initialised: Registers at 0x3f200000
[    2.166394] brd: module loaded
[    2.193593] loop: module loaded
[    2.205569] Loading iSCSI transport class v2.0-870.
[    2.217167] libphy: Fixed MDIO Bus: probed
[    2.226675] usbcore: registered new interface driver lan78xx
[    2.235865] usbcore: registered new interface driver smsc95xx
[    2.244658] dwc_otg: version 3.00a 10-AUG-2012 (platform bus)
[    2.982016] Core Release: 2.80a
[    2.990774] Setting default values for core params
[    2.999449] Finished setting default values for core params
[    3.208554] Using Buffer DMA mode
[    3.217050] Periodic Transfer Interrupt Enhancement - disabled
[    3.225674] Multiprocessor Interrupt Enhancement - disabled
[    3.234414] OTG VER PARAM: 0, OTG VER FLAG: 0
[    3.243078] Dedicated Tx FIFOs mode

[    3.253691] WARN::dwc_otg_hcd_init:1074: FIQ DMA bounce buffers: virt = 9ed04000 dma = 0xded04000 len=9024
[    3.271254] FIQ FSM acceleration enabled for :
               Non-periodic Split Transactions
               Periodic Split Transactions
               High-Speed Isochronous Endpoints
               Interrupt/Control Split Transaction hack enabled
[    3.314540] dwc_otg: Microframe scheduler enabled

[    3.314669] WARN::hcd_init_fiq:457: FIQ on core 1

[    3.331186] WARN::hcd_init_fiq:458: FIQ ASM at 807c8de0 length 36

[    3.347449] WARN::hcd_init_fiq:497: MPHI regs_base at bb810000
[    3.363518] dwc_otg 3f980000.usb: DWC OTG Controller
[    3.371758] dwc_otg 3f980000.usb: new USB bus registered, assigned bus number 1
[    3.379942] dwc_otg 3f980000.usb: irq 89, io mem 0x00000000
[    3.388129] Init: Port Power? op_state=1
[    3.396091] Init: Power Port (0)
[    3.404550] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 5.10
[    3.412601] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.420686] usb usb1: Product: DWC OTG Controller
[    3.428743] usb usb1: Manufacturer: Linux 5.10.52-v7+ dwc_otg_hcd
[    3.436948] usb usb1: SerialNumber: 3f980000.usb
[    3.446189] hub 1-0:1.0: USB hub found
[    3.454441] hub 1-0:1.0: 1 port detected
[    3.464244] dwc_otg: FIQ enabled
[    3.464266] dwc_otg: NAK holdoff enabled
[    3.464280] dwc_otg: FIQ split-transaction FSM enabled
[    3.464308] Module dwc_common_port init
[    3.464811] usbcore: registered new interface driver usb-storage
[    3.473389] mousedev: PS/2 mouse device common for all mice
[    3.483562] bcm2835-wdt bcm2835-wdt: Broadcom BCM2835 watchdog timer
[    3.494748] sdhci: Secure Digital Host Controller Interface driver
[    3.502965] sdhci: Copyright(c) Pierre Ossman
[    3.512076] sdhost-bcm2835 3f202000.mmc: could not get clk, deferring probe
[    3.520800] sdhci-pltfm: SDHCI platform and OF driver helper
[    3.530261] ledtrig-cpu: registered to indicate activity on CPUs
[    3.538927] hid: raw HID events driver (C) Jiri Kosina
[    3.547292] usbcore: registered new interface driver usbhid
[    3.555423] usbhid: USB HID core driver
[    3.570099] Initializing XFRM netlink socket
[    3.578247] NET: Registered protocol family 17
[    3.586398] Key type dns_resolver registered
[    3.595186] Indeed it is in host mode hprt0 = 00021501
[    3.603241] Registering SWP/SWPB emulation handler
[    3.611217] registered taskstats version 1
[    3.619065] Loading compiled-in X.509 certificates
[    3.664071] Key type ._fscrypt registered
[    3.672051] Key type .fscrypt registered
[    3.679775] Key type fscrypt-provisioning registered
[    3.705301] bcm2835-power bcm2835-power: Broadcom BCM2835 power domains driver
[    3.716702] sdhost: log_buf @ (ptrval) (ded03000)
[    3.773267] mmc0: sdhost-bcm2835 loaded - DMA enabled (>1)
[    3.785258] of_cfs_init
[    3.793626] of_cfs_init: OK
[    3.803214] Waiting for root device PARTUUID=6e8533f3-02...
[    3.810094] usb 1-1: new high-speed USB device number 2 using dwc_otg
[    3.819088] Indeed it is in host mode hprt0 = 00001101
[    3.871201] mmc0: host does not support reading read-only switch, assuming write-enable
[    3.883899] mmc0: new high speed SDHC card at address aaaa
[    3.893334] mmcblk0: mmc0:aaaa SU16G 14.8 GiB
[    3.907557]  mmcblk0: p1 p2
[    3.941844] EXT4-fs (mmcblk0p2): INFO: recovery required on readonly filesystem
[    3.949772] EXT4-fs (mmcblk0p2): write access will be enabled during recovery
[    4.060658] usb 1-1: New USB device found, idVendor=0424, idProduct=9514, bcdDevice= 2.00
[    4.068928] usb 1-1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    4.078333] hub 1-1:1.0: USB hub found
[    4.086661] hub 1-1:1.0: 5 ports detected
[    4.410066] usb 1-1.1: new high-speed USB device number 3 using dwc_otg
[    4.509579] EXT4-fs (mmcblk0p2): recovery complete
[    4.527713] EXT4-fs (mmcblk0p2): mounted filesystem with ordered data mode. Opts: (null)
[    4.536116] VFS: Mounted root (ext4 filesystem) readonly on device 179:2.
[    4.540548] usb 1-1.1: New USB device found, idVendor=0424, idProduct=ec00, bcdDevice= 2.00
[    4.552856] usb 1-1.1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    4.554845] devtmpfs: mounted
[    4.572643] smsc95xx v2.0.0
[    4.583241] Freeing unused kernel memory: 1024K
[    4.592150] Run /sbin/init as init process
[    4.600452]   with arguments:
[    4.600468]     /sbin/init
[    4.600482]   with environment:
[    4.600496]     HOME=/
[    4.600510]     TERM=linux
[    5.106907] libphy: smsc95xx-mdiobus: probed
[    5.116912] smsc95xx 1-1.1:1.0 eth0: register 'smsc95xx' at usb-3f980000.usb-1.1, smsc95xx USB 2.0 Ethernet, m.m.m.m
[    5.493209] systemd[1]: System time before build time, advancing clock.
[    5.598825] NET: Registered protocol family 10
[    5.609838] Segment Routing with IPv6
[    5.703230] systemd[1]: systemd 241 running in system mode. (+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN -PCRE2 default-hierarchy=hybrid)
[    5.724497] systemd[1]: Detected architecture arm.
[    5.834293] systemd[1]: Set hostname to <server>.
[    7.025309] random: systemd: uninitialized urandom read (16 bytes read)
[    7.053776] random: systemd: uninitialized urandom read (16 bytes read)
[    7.063846] systemd[1]: Listening on initctl Compatibility Named Pipe.
[    7.085200] random: systemd: uninitialized urandom read (16 bytes read)
[    7.095623] systemd[1]: Listening on udev Kernel Socket.
[    7.117598] systemd[1]: Listening on Journal Socket.
[    7.149046] systemd[1]: Mounting RPC Pipe File System...
[    7.173266] systemd[1]: Condition check resulted in Huge Pages File System being skipped.
[    7.193721] systemd[1]: Mounting POSIX Message Queue File System...
[    7.220172] systemd[1]: Created slice User and Session Slice.
[    7.395038] i2c /dev entries driver
[    8.417823] EXT4-fs (mmcblk0p2): re-mounted. Opts: (null)
[    8.614474] systemd-journald[125]: Received request to flush runtime journal from PID 1
[    9.577536] mc: Linux media interface: v0.10
[    9.610650] vc_sm_cma: module is from the staging directory, the quality is unknown, you have been warned.
[    9.615524] bcm2835_vc_sm_cma_probe: Videocore shareuser@server:/$ d memory driver
[    9.615597] [vc_sm_connected_init]: start
[    9.630380] [vc_sm_connected_init]: installed successfully
[    9.657854] videodev: Linux video capture interface: v2.00
[    9.713260] bcm2835_mmal_vchiq: module is from the staging directory, the quality is unknown, you have been warned.
[    9.713271] bcm2835_mmal_vchiq: module is from the staging directory, the quality is unknown, you have been warned.
[    9.727181] bcm2835_isp: module is from the staging directory, the quality is unknown, you have been warned.
[    9.730120] bcm2835_v4l2: module is from the staging directory, the quality is unknown, you have been warned.
[    9.736815] bcm2835_codec: module is from the staging directory, the quality is unknown, you have been warned.
[    9.757663] bcm2835-codec bcm2835-codec: Device registered as /dev/video10
[    9.757817] bcm2835-codec bcm2835-codec: Loaded V4L2 decode
[    9.759520] bcm2835-isp bcm2835-isp: Device node output[0] registered as /dev/video13
[    9.760560] bcm2835-isp bcm2835-isp: Device node capture[0] registered as /dev/video14
[    9.761461] bcm2835-isp bcm2835-isp: Device node capture[1] registered as /dev/video15
[    9.762267] bcm2835-isp bcm2835-isp: Device node stats[2] registered as /dev/video16
[    9.762341] bcm2835-isp bcm2835-isp: Register output node 0 with media controller
[    9.762399] bcm2835-isp bcm2835-isp: Register capture node 1 with media controller
[    9.762480] bcm2835-isp bcm2835-isp: Register capture node 2 with media controller
[    9.762521] bcm2835-isp bcm2835-isp: Register capture node 3 with media controller
[    9.763106] bcm2835-isp bcm2835-isp: Loaded V4L2 bcm2835-isp
[    9.766974] bcm2835-codec bcm2835-codec: Device registered as /dev/video11
[    9.767068] bcm2835-codec bcm2835-codec: Loaded V4L2 encode
[    9.788588] bcm2835-codec bcm2835-codec: Device registered as /dev/video12
[    9.788705] bcm2835-codec bcm2835-codec: Loaded V4L2 isp
[   10.515931] checking generic (3e513000 6e7000) vs hw (0 ffffffff)
[   10.515968] fb0: switching to vc4drmfb from simple
[   10.530580] Console: switching to colour dummy device 80x30
[   10.540691] vc4-drm soc:gpu: bound 3f400000.hvs (ops vc4_hvs_ops [vc4])
[   10.560704] Registered IR keymap rc-cec
[   10.561423] rc rc0: vc4 as /devices/platform/soc/3f902000.hdmi/rc/rc0
[   10.562392] input: vc4 as /devices/platform/soc/3f902000.hdmi/rc/rc0/input0
[   10.576180] vc4_hdmi 3f902000.hdmi: Could not register sound card: -517
[   10.905638] vc4-drm soc:gpu: bound 3f400000.hvs (ops vc4_hvs_ops [vc4])
[   10.906885] Registered IR keymap rc-cec
[   10.907226] rc rc0: vc4 as /devices/platform/soc/3f902000.hdmi/rc/rc0
[   10.907614] input: vc4 as /devices/platform/soc/3f902000.hdmi/rc/rc0/input1
[   10.918261] vc4-drm soc:gpu: bound 3f902000.hdmi (ops vc4_hdmi_ops [vc4])
[   10.918701] vc4-drm soc:gpu: bound 3f806000.vec (ops vc4_vec_ops [vc4])
[   10.919229] vc4-drm soc:gpu: bound 3f004000.txp (ops vc4_txp_ops [vc4])
[   10.919660] vc4-drm soc:gpu: bound 3f206000.pixelvalve (ops vc4_crtc_ops [vc4])
[   10.920184] vc4-drm soc:gpu: bound 3f207000.pixelvalve (ops vc4_crtc_ops [vc4])
[   10.920661] vc4-drm soc:gpu: bound 3f807000.pixelvalve (ops vc4_crtc_ops [vc4])
[   10.921002] vc4-drm soc:gpu: bound 3fc00000.v3d (ops vc4_v3d_ops [vc4])
[   10.926307] [drm] Initialized vc4 0.0.0 20140616 for soc:gpu on minor 0
[   11.010364] Console: switching to colour frame buffer device 240x67
[   11.047609] vc4-drm soc:gpu: [drm] fb0: vc4drmfb frame buffer device
[   13.662499] 8021q: 802.1Q VLAN Support v1.8
[   13.928917] cfg80211: Loading compiled-in X.509 certificates for regulatory database
[   14.050174] Adding 102396k swap on /var/swap.  Priority:-2 extents:1 across:102396k SSFS
[   14.064514] cfg80211: Loaded X.509 cert 'sforshee: 00b28ddf47aef9cea7'
[   14.353789] SMSC LAN8700 usb-001:003:01: attached PHY driver [SMSC LAN8700] (mii_bus:phy_addr=usb-001:003:01, irq=POLL)
[   14.354476] smsc95xx 1-1.1:1.0 eth0: hardware isn't capable of remote wakeup
[   14.371575] smsc95xx 1-1.1:1.0 eth0: Link is Down
[   16.491668] smsc95xx 1-1.1:1.0 eth0: Link is Up - 100Mbps/Full - flow control off
[   16.491746] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[   18.094591] ICMPv6: process `dhcpcd' is using deprecated sysctl (syscall) net.ipv6.neigh.eth0.retrans_time - use net.ipv6.neigh.eth0.retrans_time_ms instead
[   24.436389] random: crng init done
[   24.436419] random: 7 urandom warning(s) missed due to ratelimiting
```

**Logs**
It appears that the Pi doesn't write anything to `syslog` or `kern.log` when the boot issue occurs.

**Additional context**
I run Raspbian Buster Lite. However, this originally was a Stretch install which I upgraded after Stretch became EOL. I resized the boot partition to 256mb (although rpi-update complains it is `255M` and should be `256M` instead). I never had any problems after the upgrade until now.

My `/boot/config.txt`:
```
# For more options and information see
# http://rpf.io/configtxt
# Some settings may impact device functionality. See link above for details

# uncomment if you get no picture on HDMI for a default "safe" mode
#hdmi_safe=1

# uncomment this if your display has a black border of unused pixels visible
# and your display can output without overscan
#disable_overscan=1

# uncomment the following to adjust overscan. Use positive numbers if console
# goes off screen, and negative if there is too much border
#overscan_left=16
#overscan_right=16
#overscan_top=16
#overscan_bottom=16

# uncomment to force a console size. By default it will be display's size minus
# overscan.
#framebuffer_width=1280
#framebuffer_height=720

# uncomment if hdmi display is not detected and composite is being output
#hdmi_force_hotplug=1

# uncomment to force a specific HDMI mode (this will force VGA)
#hdmi_group=1
#hdmi_mode=1

# uncomment to force a HDMI mode rather than DVI. This can make audio work in
# DMT (computer monitor) modes
#hdmi_drive=2

# uncomment to increase signal to HDMI, if you have interference, blanking, or
# no display
#config_hdmi_boost=4

# uncomment for composite PAL
#sdtv_mode=2

#uncomment to overclock the arm. 700 MHz is the default.
#arm_freq=800

# Uncomment some or all of these to enable the optional hardware interfaces
dtparam=i2c_arm=off
#dtparam=i2s=on
#dtparam=spi=on

# Uncomment this to enable the lirc-rpi module
#dtoverlay=lirc-rpi

# Additional overlays and parameters are documented /boot/overlays/README

# Enable audio (loads snd_bcm2835)
dtparam=audio=off
start_x=0
enable_uart=0
#gpu_mem=16
[all]
#dtoverlay=vc4-fkms-v3d
[all]
dtoverlay=vc4-kms-v3d
```

**Comment by popcornmix** [comment]
I see you are using the kms driver - is there a reason for this?
It's still somewhat experimental and I believe that is the cause of the failure to boot without hdmi attached.
Remove that line (or switch to fkms) will probably avoid the issue.

It is something that should be fixed.

**Comment by rpdom** [comment]
I'm having an issue with the latest update to 64-bit kernel 5.10.52-v8+ #1440 on a Pi 3B+. It fails to boot when a monitor isn't attached.

This was a freshly created Raspi OS Buster 64-bit Lite. No additional software installed. SSH enabled.

I found that commenting out the dtoverlay=vc4-kms-v3d entry in /boot/config.txt allows the 3B+ to boot without a monitor attached. That entry is enabled by default in the downloaded image.

The issue does not occur when using a Pi 3B or Pi 400.

https://www.raspberrypi.org/forums/viewtopic.php?p=1897434#p1897434

**Comment by popcornmix** [comment]
@mripard I can reproduce this on a Pi 3B+ and a Pi3B. Boot was successful on a Pi4 without HDMI.
Attaching debugger shows bus lockup, which is typical after accessing a register without clock running.
[dmesg log](https://pastebin.com/LF8uyztq)

config.txt
```
[pi4]
max_framebuffers=2
dtoverlay=rpivid-v4l2
arm_64bit=1
arm_freq=1800
[all]
disable_overscan=1
dtparam=audio=on
dtoverlay=vc4-kms-v3d
enable_uart=1
dtoverlay=disable-bt
```
cmdline.txt
```
console=tty1 console=serial0,115200 nfsroot=10.3.14.73:/home/dom/bullseye,vers=3 rw ip=dhcp rootwait
```

**Comment by lategoodbye** [comment]
Possible duplicate of #4457 ?

**Comment by Hiradur** [comment]
>I see you are using the kms driver - is there a reason for this?
It's still somewhat experimental and I believe that is the cause of the failure to boot without hdmi attached.
Remove that line (or switch to fkms) will probably avoid the issue.

Thanks for spotting this. I played around with GPU acceleration one or two years ago and forgot to revert to the non-GL driver. I don't need the KMS driver anymore so I disabled it. I reinstalled the `raspberrypi-kernel` package and related packages to get back to the stable Kernel package (`Linux server 5.10.52-v7+ #1441 SMP Tue Aug 3 18:10:09 BST 2021 armv7l GNU/Linux`) as [suggested in the docs](https://www.raspberrypi.org/documentation/raspbian/applications/rpi-update.md). The Pi boots successfully now.
